### ISSUE TITLE
Invalid Code Point

### ISSUE THREAD

**Issue by ImVexed** [vshaxe/hashlink-debugger#31]
HashLink debugger version: 0.5.1
HashLink version: 1.8.0
Haxe version: 4.0.0-preview.5+7eb789f54
Heaps version: 1.5.0

Clone of https://github.com/vshaxe/hashlink-debugger/issues/29 that is reported fixed, potential regression?

I'm using the vscode plugin & the https://heaps.io/samples/world.html sample in a new project and upon using the vscode autogenerated HashLink (launch)/(attach) debug configs I get `RangeError: Invalid code point 4038625`

My auto-generated launch.json:
```
{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": [
        {
            "name": "HashLink (launch)",
            "request": "launch",
            "type": "hl",
            "hxml": "build.hxml",
            "cwd": "${workspaceRoot}",
            "preLaunchTask": "build"
        },
        {
            "name": "HashLink (attach)",
            "request": "attach",
            "port": 6112,
            "type": "hl",
            "hxml": "build.hxml",
            "cwd": "${workspaceRoot}",
            "preLaunchTask": "build"
        }
    ]
}
```

My tasks.json (I had to manually add the `taskName` field because `preLaunchTask` wasn't finding the task without it)

```
{
    // See https://go.microsoft.com/fwlink/?LinkId=733558
    // for the documentation about the tasks.json format
    "version": "2.0.0",
    "tasks": [
        {
            "taskName": "build",
            "type": "haxe",
            "args": "active configuration",
            "problemMatcher": [
                "$haxe-absolute",
                "$haxe",
                "$haxe-error",
                "$haxe-trace"
            ],
            "group": {
                "kind": "build",
                "isDefault": true
            }
        }
    ]
}
```

A standard `haxe build.hxml && hl world.hl` works fine

Here's my directory structure:

```
Directory of D:\haxe\helloHL

12/09/2018  00:36    <DIR>          .
12/09/2018  00:36    <DIR>          ..
12/09/2018  00:16    <DIR>          .vscode
12/09/2018  00:02                48 build.hxml
12/08/2018  23:34    <DIR>          res
12/08/2018  23:59             1,468 World.hx
               2 File(s)          1,516 bytes

 Directory of D:\haxe\helloHL\.vscode

12/09/2018  00:16    <DIR>          .
12/09/2018  00:16    <DIR>          ..
12/09/2018  00:15               769 launch.json
12/09/2018  00:16               559 tasks.json
               2 File(s)          1,328 bytes

 Directory of D:\haxe\helloHL\res

12/08/2018  23:34    <DIR>          .
12/08/2018  23:34    <DIR>          ..
12/08/2018  23:33    <DIR>          .tmp
12/08/2018  23:32             7,956 rock.hmd
12/08/2018  23:34            38,233 rockTexture.jpg
12/08/2018  23:32            15,999 tree.hmd
12/08/2018  23:34            30,174 treeTexture.png
               4 File(s)         92,362 bytes
```

**Comment by ncannasse** [comment]
The fix requires both HL and Haxe GIT.

**Comment by ImVexed** [comment]
@ncannasse And per your previous comment on the other issue Haxe GIT won't be out until 4.0-preview 6 and HL 1.9?

**Comment by ncannasse** [comment]
Yes exactly

**Comment by Gama11** [comment]
Looks like this can be closed now with HL 1.9 and Haxe 4.0.0-rc.1 released.
